### PR TITLE
Move self_bitor into the callable bit_or

### PR DESCRIPTION
**Pull request by SadiinsoSnowfall**

### CODE REVIEW

File: include/eve/arch/cpu/wide.hpp [hunk 1]
```diff
@@ -451,43 +451,41 @@ namespace eve
...
     template<scalar_value S>
-    friend EVE_FORCEINLINE wide operator|(wide const& v, S w) noexcept
+    friend EVE_FORCEINLINE wide operator|(wide const& w, S s) noexcept
 #if !defined(EVE_DOXYGEN_INVOKED)
         requires(!kumi::product_type<Type> && supports_bitwise_call<wide, S>)
 #endif
     {
-      auto that = v;
-      return that |= bit_cast(w, as<Type> {});
+      return bit_or(w, bit_cast(s, as<Type>{}));
```
Comment (DenisYaroshevskiy) [inline include/eve/arch/cpu/wide.hpp:477]: These two free operators are inconsistent with the non free one. Let's make them consistent.
Ideally you also want to call `operator|=` as an implementation of `|`

File: include/eve/module/core/regular/bit_or.hpp [hunk 2]
```diff
@@ -16,27 +16,24 @@ namespace eve
   template<typename Options>
   struct bit_or_t : strict_tuple_callable<bit_or_t, Options>
   {
-    template<eve::value T0, value T1>
-    requires(eve::same_lanes_or_scalar<T0, T1>)
-    EVE_FORCEINLINE constexpr bit_value_t<T0, T1>
-    operator()(T0 t0, T1 t1) const noexcept
+    template<value T0, value T1>
+    EVE_FORCEINLINE constexpr bit_value_t<T0, T1> operator()(T0 t0, T1 t1) const noexcept
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/bit_or.hpp:20]: I really dislike `bit_value`
It has no documentation.
This name doesn't tell me anything. 

Our documentation for bit_or is also bad and doesn't specify a return value.

This is not for this pull request, I'm just voicing me being upset.

File: include/eve/arch/cpu/wide.hpp (hunk 1 above)
Comment (DenisYaroshevskiy) [reply include/eve/arch/cpu/wide.hpp:477]: Inconsistent: there are 2 overloads of operator| while just one for operator|=

It's not a needless jump, it's a way to keep them consistent. You'd find it recommended in many places

File: include/eve/module/core/regular/bit_or.hpp (hunk 2 above)
Comment (jfalcou) [reply include/eve/module/core/regular/bit_or.hpp:20]: Let fix this in the cleanup PR

**Review by DenisYaroshevskiy — APPROVED** [review]

**Pull request merged**